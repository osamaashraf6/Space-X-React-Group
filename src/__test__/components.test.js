import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../redux/configureStore';
import DisplayRockets from '../components/rocket/displayrockets/displayrockets';
import SingleRocket from '../components/rocket/singlerocket/singlerocket';
import Header from '../components/header/header';

describe('Test header component', () => {
  it('header renders correctly', () => {
    const tree = renderer
      .create(
        <Router>
          <Header />
        </Router>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

it('Test Rocket renders correctly', () => {
  const item = {
    description: 'description',
    image: 'https://test.com',
    name: 'Rocket',
    id: 1,
    reserved: false,
  };
  const tree = renderer.create(
    <Provider store={store}>
      <DisplayRockets
        description={item.description}
        image={item.image}
        rocketName={item.name}
        id={item.id}
        reserved={item.reserved}
      />
    </Provider>,
  );
  expect(tree).toMatchSnapshot();
});

it('cdf', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <SingleRocket />
    </Provider>,
  );
  expect(tree).toMatchSnapshot();
});
