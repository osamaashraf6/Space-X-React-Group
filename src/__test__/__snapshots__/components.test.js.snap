// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Rocket renders correctly 1`] = `
<div
  className="rocket-container"
>
  <hr
    className="space-below"
  />
</div>
`;

exports[`Test Rockets renders correctly 1`] = `
<div
  className="rocket-container"
>
  <hr
    className="space-below"
  />
</div>
`;

exports[`Test header component header renders correctly 1`] = `
<nav
  className="nav"
  id="nav"
>
  <div
    className="container"
  >
    <div
      className="nav__container__parnav"
    >
      <div
        className="nav__container__parnav__brand"
      >
        <a
          className="nav__container__parnav__brand__logo"
          href="#logo"
        >
          <img
            alt="logo-top"
            className="nav__container__parnav__brand__logo__img"
            src="planet2.png"
          />
          <h3>
            Space Traveler's Hub
          </h3>
        </a>
      </div>
      <ul
        className="nav__container__parnav__ul"
      >
        <li>
          <a
            className=""
            href="/rocket"
            onClick={[Function]}
          >
            Rockets
          </a>
        </li>
        <li>
          <a
            className=""
            href="/mission"
            onClick={[Function]}
          >
            mission
          </a>
        </li>
        <hr />
        <li>
          <a
            className=""
            href="/profile"
            onClick={[Function]}
          >
            profile
          </a>
        </li>
      </ul>
    </div>
  </div>
</nav>
`;

exports[`cdf 1`] = `
<div
  className="rocket-data"
>
  <div>
    <picture>
      <img
        className="image"
      />
    </picture>
  </div>
  <div
    className="description"
  >
    <div
      className="rocket-name"
    >
      <p
        className="bird-name"
      />
    </div>
    <div
      className="rocket-description"
    >
      <p />
    </div>
    <div
      className="reserve-action"
    >
      <button
        className="btnReservation"
        onClick={[Function]}
        type="button"
      >
        Reserve Rocket
      </button>
    </div>
  </div>
</div>
`;
